presto-hive/src/test/java/com/facebook/presto/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -166,17 +166,21 @@ public void testPathFilterBucketedPartitionedTable()
     private List<String> drain(HiveSplitSource source)
             throws Exception
     {
-        ImmutableList.Builder<String> paths = ImmutableList.builder();
-        while (true) {
-            List<ConnectorSplit> splits = source.getNextBatch(100).get();
-            for (ConnectorSplit connectorSplit : splits) {
-                paths.add(((HiveSplit) connectorSplit).getPath());
-            }
-            if (source.isFinished()) {
-                break;
-            }
+        return drainSplits(source).stream()
+                .map(HiveSplit::getPath)
+                .collect(toImmutableList());
+    }
+
+    private List<HiveSplit> drainSplits(HiveSplitSource source)
+            throws Exception
+    {
+        ImmutableList.Builder<HiveSplit> splits = ImmutableList.builder();
+        while (!source.isFinished()) {
```

Comment:
IMO your refactor changed the semantic. Previously you `do-while` loop, now you have a `while`. Previously `paths` where added even for first finished source.

Reply:
Consciously. I will add note to cmt msg.